Update install instructions (#644)
CLIMADA-project/climada_python: doc/guide/install.rst

* Create a **workspace directory**, ideally in subdirectory of your user/home directory.
  Do **not** use a directory that is synchronized by cloud storage systems like OneDrive, iCloud or Polybox!

.. hint:: If you need help with the vocabulary used on this page, refer to the :ref:`Glossary <install-glossary>`.

.. _install-choice:

Decide on Your Entry Level!
^^^^^^^^^^^^^^^^^^^^^^^^^^^

Depening on your level of expertise, we provide two different approaches:

* If you have never worked with a command line, or if you just want to give CLIMADA a try, follow the :ref:`simple instructions <install-simple>`.
* If you want to use the very latest development version of CLIMADA or even develop new CLIMADA code, follow the :ref:`advanced instructions <install-advanced>`.
  If you want to install `CLIMADA Petals`_, also follow these.

Both approaches are not mutually exclusive.
After successful installation, you may switch your setup at any time.

.. _install-simple:

-------------------
Simple Instructions
-------------------

These instructions will install the most recent stable version of CLIMADA without cloning its repository.

> #. Download the Anaconda environment specifications for CLIMADA: :download:`env_climada.yml </../requirements/env_climada.yml>`.

Author: I see. I am a bit hesitant because I do not like instructing people to "mindlessly" copy-paste command line instructions, as they can be quite harmful. The other thing I was thinking about is the availability of `curl`. Can we safely assume that it is available in all setups we encourage? Or do Linux users first have to `apt install curl`?

> #. Download the Anaconda environment specifications for CLIMADA: :download:`env_climada.yml </../requirements/env_climada.yml>`.

Reviewer: Actually the availability was the argument for using `curl` and not `wget`: `curl` is available on Windows and macOS.
I assume the availability on Linux depends on the distribution. But I guess the number of Linux users that don't have `curl` installed is neglectable. 🧐

> #. Download the Anaconda environment specifications for CLIMADA: :download:`env_climada.yml </../requirements/env_climada.yml>`.

Author: Then I'll use the `curl` instruction and add to the prerequisites that Linux users have to install `curl`, `python` and `git` 😁

>    You can also find it online in the `CLIMADA GitHub repository <https://github.com/CLIMADA-project/climada_python/blob/main/requirements/env_climada.yml>`_.

Author: Alright! I just realize that I should add to this that users need to download the file into the workspace directory

   Move or copy the file into your workspace directory.

> #. Open the command line and navigate to the workspace directory you created.

Reviewer: Oops. No. Sorry. Somehow just missed it.

   Executing CLIMADA for the first time will take some time because it will generate a directory tree in your home/user directory.
   After a while, some text should appear in your terminal.
   In the end, you should see an "Ok".
   If so, great! You are good to go.

.. _install-advanced:

---------------------
Advanced Instructions
---------------------

For advanced Python users or developers of CLIMADA, we recommed cloning the CLIMADA repository and installing the package from source.

#. Open the command line and navigate to the workspace directory you created using ``cd``.
   Replace ``<path/to/workspace>`` with the path of the directory that contains the workspace folder:

   .. code-block:: shell

      cd <path/to/workspace>

#. Clone CLIMADA from its `GitHub repository <https://github.com/CLIMADA-project/climada_python>`_.
   Enter the directory and check out the branch of your choice.
   The latest development version will be available under the branch ``develop``.

   .. code-block:: shell

        git clone https://github.com/CLIMADA-project/climada_python.git
        cd climada_python
        git checkout develop

> #. Create a Anaconda environment called ``climada_env`` for installing CLIMADA.

Reviewer: ```suggestion
#. Create an Anaconda environment called ``climada_env`` for installing CLIMADA.
```

^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

Conda might report a permission error on macOS Mojave.
Carefully follow these instructions: https://github.com/conda/conda/issues/8440#issuecomment-481167572

No ``impf_TC`` Column in ``GeoDataFrame``
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

This may happen when a demo file from CLIMADA was not updated after the change in the impact function naming pattern from ``if_`` to ``impf_`` when `CLIMADA v2.2.0 <https://github.com/CLIMADA-project/climada_python/releases/tag/v2.2.0>`_ was released.
Execute

.. code-block:: shell

   conda activate climada_env
   python -c "import climada; climada.setup_climada_data(reload=True)"

.. _install-glossary:

------------------------
The What Now? (Glossary)
------------------------

You might have become confused about all the names thrown at you.
Let's clear that up:

Terminal, Command Line
    A text-only program for interacting with your computer (the old fashioned way).

`Anaconda`_, conda
    The program that installs all requirements and creates a suitable environment for CLIMADA.

Environment (Programming)
    A setup where only a specific set of modules and programs can interact.

>     This is especially useful if you want to install programs with incompatible requirements.

Reviewer: ```suggestion
    This is especially useful if you want to install programs with mutually incompatible requirements.
```

`pip <https://pip.pypa.io/en/stable/index.html>`_
    The Python package installer.

`git <https://git-scm.com/>`_

>     A program for version controlling your software (or any text-based set of files).

Reviewer: ```suggestion
    A popular version control software for programming code (or any text-based set of files).
```
